cmd/horcrux/cmd/cosigner.go

```diff
@@ -55,102 +77,117 @@ func StartCosignerCmd() *cobra.Command {
 					Nodes:             config.Nodes(),
 					Cosigners:         config.CosignerPeers(),
 				}
-			)
-
-			if _, err = os.Stat(config.PrivValKeyFile); os.IsNotExist(err) {
-				return fmt.Errorf("private key share doesn't exist at path(%s)", config.PrivValKeyFile)
 			}
 
-			logger.Info("Tendermint Validator", "mode", config.Mode, "priv-key", config.PrivValKeyFile, "priv-state-dir", config.PrivValStateDir)
-
-			key, err := signer.LoadCosignerKey(config.PrivValKeyFile)
-			if err != nil {
-				return fmt.Errorf("error reading cosigner key: %s", err)
+			if _, err = os.Stat(cfg.PrivValKeyFile); os.IsNotExist(err) {
+				return fmt.Errorf("private key share doesn't exist at path(%s)", cfg.PrivValKeyFile)
 			}
 
-			// ok to auto initialize on disk since the cosigner share is the one that actually
-			// protects against double sign - this exists as a cache for the final signature
-			stateFile := path.Join(config.PrivValStateDir, fmt.Sprintf("%s_priv_validator_state.json", chainID))
-			signState, err := signer.LoadOrCreateSignState(stateFile)
-			if err != nil {
-				panic(err)
-			}
+			logger.Info("Tendermint Validator", "mode", cfg.Mode, "priv-key", cfg.PrivValKeyFile, "priv-state-dir", cfg.PrivValStateDir)
 
-			// state for our cosigner share
-			// Not automatically initialized on disk to avoid double sign risk
-			shareStateFile := path.Join(config.PrivValStateDir, fmt.Sprintf("%s_share_sign_state.json", chainID))
-			shareSignState, err := signer.LoadSignState(shareStateFile)
-			if err != nil {
-				panic(err)
-			}
+			var val types.PrivValidator
+			if single {
+				stateFile := path.Join(cfg.PrivValStateDir, fmt.Sprintf("%s_priv_validator_state.json", chainID))
 
-			cosigners := []signer.Cosigner{}
-			remoteCosigners := []signer.RemoteCosigner{}
+				// TODO either stop creating state file at config init
```

Comment:
Either stop creating state file at config init, or document the hell out of this. Signer cluster will sign whatever block the node throws at it so state file is only protection on this side from doublesign. Not creating on init and having a separate command tree would raise user awareness of the potential doublesign dangers here. Commands for management of this state are nice to haves as well.

Reply:
State file will still need to be generated with config init, for now. Adding commands to fetch and manage state is a good idea, something to make note of and get done in another PR.